File: okj/rendszeruzemelteto/eucsatlakozas/c#/Csatlakozas.cs
using System;

public class Csatlakozas {

    public readonly string orszag;
    public readonly DateTime datum;

    public Csatlakozas(string line) {
        var split = line.Split(';');

        orszag = split[0];
        datum = DateTime.Parse(split[1]);
    }
}

File: okj/rendszeruzemelteto/eucsatlakozas/java/Csatlakozas.java
import java.time.*;

public class Csatlakozas {

    public final String orszag;
    public final LocalDate datum;

    public Csatlakozas(String line) {
        var split = line.split(";");

        orszag = split[0];
        datum = LocalDate.parse(split[1].replace('.', '-'));
    }
}